modules/programs/bacon.nix

```
@@ -0,0 +1,38 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.bacon;
+
+  package = pkgs.bacon;
+  settingsFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ maintainers.shimunn ];
+
+  options.programs.bacon = {
+    enable = mkEnableOption "background rust code check";
+
+    preferences = mkOption {
```

This option should be called `settings` as per RFC-42:
```suggestion
    settings = mkOption {
```